File: app/src/main/java/pro/games_box/recycleradapter/MainActivity.java
package pro.games_box.recycleradapter;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    // Список с данными, которые хотим отобразить
    public List<RecyclerAdapterItem> data = new ArrayList<>();
    // Ресакйлер который будет отображить список
    public RecyclerView main_recycler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initList();
        main_recycler = (RecyclerView) findViewById(R.id.list);

        // Инициализируем ресайклер
        main_recycler.setHasFixedSize(true);
        LinearLayoutManager llm = new LinearLayoutManager(this);
        main_recycler.setLayoutManager(llm);

        // Назначаем ему адаптер
        RecyclerAdapter adapter = new RecyclerAdapter(this, data);
        main_recycler.setAdapter(adapter);

    }

    public void initList(){
        // Набиваем список данными для примера
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news1)[0],
                getResources().getStringArray(R.array.news1)[1]));
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news2)[0],
                getResources().getStringArray(R.array.news2)[1]));
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news3)[0],
                getResources().getStringArray(R.array.news3)[1]));
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news4)[0],
                getResources().getStringArray(R.array.news4)[1]));
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news5)[0],
                getResources().getStringArray(R.array.news5)[1]));
        data.add(new RecyclerAdapterItem(getResources().getStringArray(R.array.news6)[0],
                getResources().getStringArray(R.array.news6)[1]));
    }
}

File: app/src/main/java/ru/abstractfactory/recycleradapter/MainActivity.kt
package ru.abstractfactory.recycleradapter

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import ru.abstractfactory.recycleradapter.data.ItemNews
import ru.abstractfactory.recycleradapter.fragment.NewsFragment
import java.util.ArrayList


class MainActivity : AppCompatActivity() {

    companion object {
        var currentPosition = 0
        const val KEY_CURRENT_POSITION = "KEY_CURRENT_POSITION"
    }

    var data: ArrayList<ItemNews> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initList()

        if (savedInstanceState != null) {
            currentPosition = savedInstanceState.getInt(KEY_CURRENT_POSITION, 0)
            return
        }
        val fragmentManager = supportFragmentManager
        fragmentManager
                .beginTransaction()
                .add(R.id.container, NewsFragment(), NewsFragment.TAG)
                .commit()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt(KEY_CURRENT_POSITION, currentPosition)
    }

    private fun initList() {
        data.add(ItemNews(
                resources.getStringArray(R.array.news1)[0],
                resources.getStringArray(R.array.news1)[1],
                resources.getStringArray(R.array.news1)[2]))
        data.add(ItemNews(
                resources.getStringArray(R.array.news2)[0],
                resources.getStringArray(R.array.news2)[1],
                resources.getStringArray(R.array.news2)[2]))
        data.add(ItemNews(
                resources.getStringArray(R.array.news3)[0],
                resources.getStringArray(R.array.news3)[1],
                resources.getStringArray(R.array.news3)[2]))
        data.add(ItemNews(
                resources.getStringArray(R.array.news4)[0],
                resources.getStringArray(R.array.news4)[1],
                resources.getStringArray(R.array.news4)[2]))
        data.add(ItemNews(
                resources.getStringArray(R.array.news5)[0],
                resources.getStringArray(R.array.news5)[1],
                resources.getStringArray(R.array.news5)[2]))
        data.add(ItemNews(
                resources.getStringArray(R.array.news6)[0],
                resources.getStringArray(R.array.news6)[1],
                resources.getStringArray(R.array.news6)[2]))
    }
}
